analytic_engine/src/table/data.rs

```diff
@@ -480,22 +489,45 @@ impl TableData {
         should_flush
     }
 
-    /// Set `last_file_id`, mainly used in recover
-    ///
-    /// This operation require external synchronization
-    pub fn set_last_file_id(&self, last_file_id: FileId) {
-        self.last_file_id.store(last_file_id, Ordering::Relaxed);
+    /// Use allocator to alloc a file id for a new file.
+    pub async fn alloc_file_id(&self, manifest: &ManifestRef) -> GenericResult<FileId> {
+        //Persist next max file id to manifest.
+        let persist_max_file_id = move |next_max_file_id| async move {
+            self.persist_max_file_id(manifest, next_max_file_id).await
+        };
+
+        self.allocator.alloc_id(persist_max_file_id).await
     }
 
-    /// Returns the last file id
-    pub fn last_file_id(&self) -> FileId {
-        self.last_file_id.load(Ordering::Relaxed)
+    async fn persist_max_file_id(
+        &self,
+        manifest: &ManifestRef,
+        next_max_file_id: FileId,
+    ) -> GenericResult<()> {
+        let manifest_update = AllocSstIdMeta {
+            space_id: self.space_id,
+            table_id: self.id,
+            max_file_id: next_max_file_id,
+        };
+        let edit_req = {
+            let meta_update = MetaUpdate::AllocSstId(manifest_update);
+            MetaEditRequest {
+                shard_info: self.shard_info,
+                meta_edit: MetaEdit::Update(meta_update),
+            }
+        };
+        manifest.apply_edit(edit_req).await?;
```

Comment:
Add comments here to explain the `self.max_file_id` will be updated when apply this meta update.